src/mame/misc/toysoldier.cpp

```diff
@@ -0,0 +1,148 @@
+// license:GPL-2.0+
+// copyright-holders:flama12333
+/*************************************************************************
+known undumped games
+W&P CATCHER - lucky dog 
+https://youtu.be/e1Gp7Cr6KIQ
+Fantasy World - Diferent version
+https://youtu.be/I7l6sOW40gE
+there are 4 langauge Not configurable.
+chinese:
+https://youtu.be/RNM4aPQNWhg
+spanish:
+https://youtu.be/na70qV8awqE
+english female:
+https://youtu.be/wHEyHoKZkJk
+english male:
+https://youtu.be/aArV8y5TuzM?t=13
+amusements 
+https://youtu.be/qUiQgHpiAyI
+*/
+#include "emu.h"
+#include "cpu/mcs51/mcs51.h"
+#include "sound/ay8910.h"
+#include "sound/ymopl.h"
+#include "speaker.h"
+
+namespace {
+
+class toysoldier_state : public driver_device
+{
+public:
+	toysoldier_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		
+	{ }
+
+	void toysoldier(machine_config &config);
+
+private:
+	void ay_port_a_w(uint8_t data);
+	void ay_port_b_w(uint8_t data);
+	void io_map(address_map &map);
+	void program_map(address_map &map);
+
+	virtual void machine_start() override;
+
+
+};
+
+static INPUT_PORTS_START( toysoldier )
+    PORT_START("IN1") //  Gantry. 
+    PORT_BIT(0x08, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_1)
+	PORT_BIT(0x10, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_2)
+	PORT_BIT(0x20, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_CODE(KEYCODE_3)
+    PORT_START("IN2")
```

Comment:
for better readability, please consider skipping a line between each PORT_START block